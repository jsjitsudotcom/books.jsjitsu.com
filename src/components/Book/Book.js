import React, { PureComponent } from "react";
import Style from "./Book.scss";
import PropTypes from "prop-types";
import Cover from "./components/Cover/Cover";

const Loading = () => (
  <div className={Style.container}>
    <div className={Style.loadingWrapper}>
      <div className={Style.loading} />
    </div>
  </div>
);

export default class Book extends PureComponent {
  render() {
    if (this.props.loading) return <Loading />;

    return (
      <div onClick={this.props.onClick} className={Style.container}>
        <Cover cover={this.props.cover} />

        <div className={Style.info}>
          <div className={Style.title}>{this.props.title}</div>
          <div className={Style.author}>{this.props.author}</div>
          <div className={Style.date}>{this.props.date}</div>
        </div>
      </div>
    );
  }
}

Book.propTypes = {
  title: PropTypes.string,
  author: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  date: PropTypes.number,
  onClick: PropTypes.func
};

Book.defaultProps = {
  onClick: /* istanbul ignore next */ () => false
};
