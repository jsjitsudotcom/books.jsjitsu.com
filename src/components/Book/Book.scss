:local(.container) {
  width: 150px;
  padding: 0 5px;
}

:local(.cover) {
  width: 100%;
  border-radius: 4px;
  background-size: cover;
  background-position: 50%;
  background-repeat: no-repeat;
}

:local(.coverSize){
  padding-top: 150%;
}

:local(.info) {
  margin-top: 10px;
  font-size: 16px;
}

:local(.title) {
  font-weight: 500;
}

:local(.author) {
  font-size: 14px;
  font-weight: 300;
  margin-top: 5px;
}

:local(.date) {
  font-size: 14px;
  font-weight: 300;
  margin-top: 5px;
}
