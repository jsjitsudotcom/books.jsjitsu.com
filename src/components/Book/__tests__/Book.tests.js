import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import Book from "./../Book";

describe("<Book />", () => {
  it("Le componsant doit Ãªtre rendu", () => {
    shallow(<Book />);
  });

  describe("Suite de tests de l'affichage", () => {
    test("Le composant doit bien s'afficher", () => {
      const title = "title";
      const author = "author";
      const date = 1546;
      const cover = "cover";

      const wrapper = shallow(
        <Book title={title} author={author} date={date} cover={cover} />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
