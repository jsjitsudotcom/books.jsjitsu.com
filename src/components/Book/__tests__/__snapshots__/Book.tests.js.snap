// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book /> Suite de tests de l'affichage Le composant doit bien s'afficher 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <Cover
    cover="cover"
  />
  <div
    className="info"
  >
    <div
      className="title"
    >
      title
    </div>
    <div
      className="author"
    >
      author
    </div>
    <div
      className="date"
    >
      1546
    </div>
  </div>
</div>
`;
